### DISCUSSION TITLE
Berry Scripting memory limits

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#19494] [Q&A]
I'm currently using Berry scripting intensively. So the following questions arose.

What are the memory limits in berry scripting?

How can I get information about how much memory is still free or used ?

Thank you for your support.

John

**Comment by sfromis** [comment]
Memory limits varies by hardware and other features using RAM. Different ESP32 chips have variations in RAM size, and you can use boards with PSRAM to greatly expand what's available to Berry.

You can see some data about memory on the "Information" page, in the STATE payloads published, the output of the command `Status 4`, and using Berry functions `tasmota.memory()`,  `tasmota.get_free_heap()` and `tasmota.gc()`

**Comment by wjohn007** [comment]
Thank you for the hints
